lib/spack/spack/cmd/tags.py

```diff
@@ -0,0 +1,90 @@
+# Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+# Spack Project Developers. See the top-level COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+import sys
+
+import six
+
+import llnl.util.tty.colify as colify
+
+import spack.repo
+import spack.tag
+
+description = "list package tags and or their packages"
+section = "basic"
+level = "long"
+
+
+def report_tags(category, tags):
+    buffer = six.StringIO()
+    isatty = sys.stdout.isatty()
+
+    if isatty:
+        buffer.write("{0} package tags:\n".format(category))
+
+    if tags:
+        colify.colify(tags, output=buffer, tty=isatty, indent=4)
+    else:
+        buffer.write("    None\n")
+    print(buffer.getvalue())
+
+
+def setup_parser(subparser):
+    subparser.epilog = "Processes available tags when called without tags."
+    subparser.add_argument(
+        '-i', '--installed', action='store_true', default=False,
+        help="limit tags to those of installed packages"
+    )
+    subparser.add_argument(
+        '-s', '--show-packages', action='store_true', default=False,
+        help="show packages for available tags when no tags are provided"
+    )
+    subparser.add_argument(
+        'tag',
+        nargs='*',
+        help="find packages that provide this tag"
```

Comment:
Nitpick, for consistency with other help strings:
```suggestion
        help="show packages that provide this tag"
```